lib/early_ack_tracker.c

```diff
@@ -70,6 +70,14 @@ early_ack_tracker_manage_msg_ack(AckTracker *s, LogMessage *msg, AckType ack_typ
   log_pipe_unref((LogPipe *)self->super.source);
 }
 
+static void
+early_ack_tracker_free(AckTracker *s)
+{
+  EarlyAckTracker *self = (EarlyAckTracker *)s;
+
+  g_free(self);
+}
```

Comment:
g_free() is usually part of the inline function, so that allocation policy is not encoded in the free function.

Reply:
indeed...

Reply:
that's why we usually have _new() functions (which contains the g_new call) and in case we want to change the allocation policy, we also have an init_instance() function, which does initialization in the face of an existing memory block.

the existing convention is to put the g_free() in the inline function, this way it's less likely to be forgotten as further derived classes are added (as the free is always called regardless of the virtual function).

we could change the convention, but I don't think it is better the other way.

Reply:
@bazsi : I agree when we are talking about base classes/abstract classes, but not in case of interfaces...  there is no data member in the interface which means that every allocations for data members need to be done in the `init` functions. But it's just a matter of taste, I think (as we are talking about mimicking OOP in C).